src/transport/smtp/pool/async_impl.rs

```diff
@@ -134,10 +146,36 @@ impl<E: Executor> Pool<E> {
         pool
     }
 
+    pub async fn shutdown(&self) -> Result<(), Error> {
+        if let Some(connections) = self.connections.lock().await.take() {
```

Comment:
This would also be a good place to shutdown `handle`.

Reply:
I agree; however, with the current setup this would require moving the handle from an `OnceLock` to a full-blown `Mutex`, which I have decided not to implement initially for the same reason as not shutting down the sync thread - the task will immediately exit upon waking up, and pretty much consumes no resources while in this "zombie" state. If this is a worthwhile tradeoff for you though then I'll implement this change as part of the next round of feedback.

Reply:
Ah right another instance of `async-std` ruining things for us :roll_eyes:. tokio's `JoinHandle::abort` takes '&self`, async-std `JoinHandle::cancel` takes `self`. We could use [`futures_util::future::AbortHandle`](https://docs.rs/futures-util/0.3.31/futures_util/future/struct.AbortHandle.html) to get the `&self` API for both.

Reply:
Resolved in latest push